src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -174,38 +168,49 @@ def q_impl(
         alpha = self.attrs.get("alpha", 1)
         beta = self.attrs.get("beta", 1)
 
+        # If self.constant_inputs is empty this is an encrypted gemm
+        is_encrypted_gemm = isinstance(self.constant_inputs, dict) and not self.constant_inputs
+
         # If alpha != 1 or beta not in [0, 1], this function must be modified
         assert_true(alpha == 1)
-        assert_true(beta in [0, 1])
+        assert_true(beta in {0, 1})
 
         prepared_inputs = self._prepare_inputs_with_constants(
             *q_inputs, calibrate=False, quantize_actual_values=True
         )
 
-        q_input: QuantizedArray = prepared_inputs[0]
-        q_weights: QuantizedArray = prepared_inputs[1]
-        q_bias: Optional[QuantizedArray] = (
-            None if len(prepared_inputs) == 2 or beta == 0 else prepared_inputs[2]
-        )
+        q_input = prepared_inputs[0]
+        assert isinstance(q_input, QuantizedArray)
+        q_input2 = prepared_inputs[1]
+        assert isinstance(q_input2, QuantizedArray)
+        q_bias = None if len(prepared_inputs) == 2 or beta == 0 else prepared_inputs[2]
```

Comment:
maybe this is worth a comment here, more particularly : why is `or beta == 0 ` needed here ?

Reply:
This was there before. Not sure what this means without looking deeper.

Reply:
maybe @andrei-stoian-zama knows ?

Reply:
Y = alpha * A' * B' + beta * C

Thus why we need beta == 0

Reply:
ah great, worth to add it n a comment here then

Reply:
Added a comment

Reply:
great thanks  !